Source/Surfaces/PrincipalCurvature.f90

```diff
@@ -0,0 +1,267 @@
+!
+!////////////////////////////////////////////////////////////////////////
+!
+!      PrincipalCurvature.f90
+!      Created: March 19, 2025 at 10:54 AM 
+!      By: David Kopriva  
+!
+!////////////////////////////////////////////////////////////////////////
+!
+   Module PrincipalCurvatureModule
+   USE SMConstants
+   IMPLICIT NONE  
+! 
+!-----------------------------------------------------------------------
+! Two procedures are included with which to compute the maximum 
+! absolute value ofthe principal curvature at a point (x,y) 
+! of a surface z = f(x,y). 
+!
+! (1) PrincipalCurvature(gradF, HessF)
+!         REAL(KIND=RP) :: gradF(2)
+!         REAL(KIND=RP) :: hessF(2,2)
```

Comment:
```suggestion
!         REAL(KIND=RP) :: HessF(2,2)
```

Reply:
Leaving this as is since variables and functions (which can be used as variables) start with lower case letters.